_posts/2022-mm-dd-crypto-1/2022-mm-dd-crypto1-history-and-basics.markdown

```diff
@@ -0,0 +1,157 @@
+
+##### Caesar cipher
+You most probably know this one, maybe by another name. The story goes that [Gaius Julius](https://en.wikipedia.org/wiki/Julius_Caesar) needed to come up with a means of securely getting military orders to his generals. And what he came up with is a solution as simple as smart, considering the resources he reasonably might've had at hand: he shifted the letters in the alphabet. The most commonly known representation of this is that of a rotating disk marked with two alphabets. This disk allows you to quickly shift the letters as needed.
+
+//PLACEHOLDER: image of Caesar cipher disk
+
+The key in this case is the amount of letters you shift. Assuming that `A` can be represented by the number `0`, `B` by the number `1` and so on (you can see where this is going), you can for example use a key of `5` to tell the recipient of your message that you've replaced all `A`s (`0`) with `F`s (`5`), all `B`s with `G`s, etc...
+Side note: you'll see why we start counting at `0` in the example below. It just makes life a little easier.
+
+To decrypt the message, all your recipient has to do is subtract your key from the cipher text, and they'll end up with your original message. Coming back to the notation above, this means that
+> $P$ is a plain text written in the standard latin alphabet (remember the guidelines from the beginning of this post).
+> $C$ is a cipher text, also written in the latin alphabet
+> $K$ is a number that specifies the shift (I'll dive into this in a bit)
+> $e_k$ is shifting the alphabet by a number of steps defined by $K$
+> $d_k$ is shifting the alphabet by a number of steps defined as $-K$ (to invert e_k)
+
+Don't worry, we're almost there. One final problem presents itself. Assume you receive the following cipher text and a key of `5`: `HWDUYTLWFUMD`. Easy enough to do, you convert it into numbers and subtract `5`. But wait, `D` would be represented by `3`... wouldn't that make the original letter `-2`? This is where the final math part for today comes into play. A method called "modulo":
+> The latin alphabet can be represented by the modulo ring of $\mod{26}$
+> Any number within a modulo ring must be smaller than the modulo value given. E.g., $50 \mod{26} = 24$, because $50 - 26 = 24$.
+
+Remember calculating with remainders? Basically that. But kinda like the disk above, also forming a circle. The latin alphabet has 26 letters, no more, no less. And therefore, if you need to go before `A`, you go back to `Z` and continue your journey from there (and the other way around if you need to go beyond `Z`). So our `-2` becomes...? You guessed it, `Y`. With this little trick, we can reconstruct the original plain text: `CRYPTOGRAPHY`.
```

Comment:
I think the typical clock-analogy is easier to grasp, as everyone calculates modulo 24 there.